README.md

```diff
@@ -1,13 +1,219 @@
+kind: Pod
+metadata:
+  name: rdma-test-pod
+  annotations:
+    k8s.v1.cni.cncf.io/networks: sriov-rdma-net
+spec:
+  containers:
+    - name: rdma-app
+      image: centos/tools
+      imagePullPolicy: IfNotPresent
+      command: [ "/bin/bash", "-c", "--" ]
+      args: [ "while true; do sleep 300000; done;" ]
+      resources:
+        requests:
+          mellanox.com/sriov_rdma: '1'
+        limits:
+          mellanox.com/sriov_rdma: '1'
+```
+
+## SR-IOV Network Device Plugin ConfigMap example
+The following `yaml` defines an RDMA enabled SR-IOV resource pool named: `mellanox.com/sriov_rdma`
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: sriovdp-config
+  namespace: kube-system
+data:
+  config.json: |
+    {
+      "resourceList": [
+        {
+           "resourcePrefix": "mellanox.com",
+           "resourceName": "sriov_rdma",
+           "isRdma": true,
+           "selectors": {
+               "vendors": ["15b3"],
+               "pfNames": ["enp4s0f0"]
+           }
+        }
+      ]
+    }
+```
+
+## Network CRD example
+
+The following `yaml` defines a network, `sriov-network`, associated with an rdma enabled resurce, `mellanox.com/sriov_rdma`.
+
+The CNI plugins that will be executed in a chain are for Pods that request this network are: _sriov_, _rdma_ CNIs
+
+```yaml
+apiVersion: "k8s.cni.cncf.io/v1"
+kind: NetworkAttachmentDefinition
+metadata:
+  name: sriov-rdma-net
+  annotations:
+    k8s.v1.cni.cncf.io/resourceName: mellanox.com/sriov_rdma
+spec:
+  config: '{
+             "cniVersion": "0.3.1",
+             "name": "sriov-rdma-net",
+             "plugins": [{
+                          "type": "sriov",
+                          "ipam": {
+                            "type": "host-local",
+                            "subnet": "10.56.217.0/24",
+                            "routes": [{
+                              "dst": "0.0.0.0/0"
+                            }],
+                            "gateway": "10.56.217.1"
+                          }
+                        },
+                        {
+                          "type": "rdma"
+                        }]
+           }'
+```
+
+# Development
+
+### Build from source
+
+```console
+~$ git clone https://github.com/Mellanox/rdma-cni.git
+~$ cd rdma-cni
+~$ make
+running gofmt...
+running golint...
+Building rdma...
+Done!
+```
+Upon a successfull build, `rdma` binary can be found under `./build`.
+For small deployments (e.g a kubernetes test cluster/AIO K8s deployment) you can:
+1. Copy `rdma` binary to the CNI dir in each worker node.
+2. Build container image, push to your own image repo then modify the deployment template and deploy. 
 
-## At high level
-RDMA CNI plugin is intended to be run as a chained CNI plugin (introduced in CNI Specifications `v0.3.0`) moving the associated
-RDMA interfaces of a given network interface to the provided network namespace path
+#### Run tests:
+```console
+~$ make tests
+running gofmt...
```

Comment:
I prefer not to put the console output, because if we change it we probably for get to change it here

Reply:
will change